src/Storage/FileSystem/FileSystemCacheAdapter.php

```diff
@@ -90,19 +90,40 @@ public function has($key)
      *
      * @return mixed The value of the item from the cache, or $default in case of cache miss.
      *
-     * @todo throws \Psr\SimpleCache\InvalidArgumentException
+     * @throws \Psr\SimpleCache\InvalidArgumentException
      *       MUST be thrown if the $key string is not a legal value.
      */
     public function get($key, $default = null)
     {
-        $path = $this->hashDirectory->getFullPath($key);
+        $path = $this->getKeyPath($key);
 
         if (!$this->isValidFile($path)) {
             return $default;
         }
 
         /** @noinspection PhpIncludeInspection */
-        return include $path;
+        $data = include $path;
+
+        if (!$data || !\is_array($data) || !isset($data['ttl'], $data['item'])) {
```

Comment:
we don't need `\` in is_array I think.

Reply:
It is a good practice to use the FQN on global functions cos the original class may be refactored and moved to non-root namespace.